appcelerator/titanium_mobile
# fix(android): TabGroup without badge crashes when using AppCompat theme as of 9.3.0

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28189

**Summary:**
`TabGroup` should only throw an exception if using `Theme.AppCompat` and you also assign the "badge" property. This is because badges requires an app to use the `Theme.MaterialComponents` theme (which is the default). It should **NOT** throw an exception if the "badge" property is **NOT** set for backward compatibility, which is the regression this PR fixes.

**Test:**
1. Create a Classic app project from template.
2. Add the below `<application/>` entry into the "tiapp.xml".
3. Copy the below `app.js` to the project.
4. Build and run the below on Android.
5. Verify app does **NOT** crash and a badge is **NOT** shown. _(This is the fix.)_
6. Uncomment line: `tabGroupSettings.style = Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION;`
7. Build and run on Android.
8. Verify app does **NOT** crash and a badge is **NOT** shown. _(This is the fix.)_
9. Uncomment line: `tabGroup.tabs[1].badge = "5";`
10. Build and run on Android.
11. Verify app does crash with below exception. _(This is okay.)_
`The style on this component requires your app theme to be Theme.MaterialComponents`
12. Comment-out line: `tabGroupSettings.style = Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION;`
13. Build and run on Android.
14. Verify app does crash with below exception. _(This is okay.)_
`The style on this component requires your app theme to be Theme.MaterialComponents`
15. In "tiapp.xml", delete the `<application/>` XML element.
16. Build and run on Android.
17. Verify app starts fine and top tab bar shows a badge of `5`.
18. Uncomment line: `tabGroupSettings.style = Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION;`
19. Build and run on Android.
20. Verify app starts fine and bottom tab bar shows a badge of `5`.

_tiapp.xml_
```xml
<ti:app>
	<android xmlns:android="http://schemas.android.com/apk/res/android">
		<manifest>
			<application android:theme="@style/Theme.AppCompat"/>
		</manifest>
	</android>
</ti:app>
```

_app.js_
```javascript
function createTab(title) {
	var window = Ti.UI.createWindow({ title: title });
	window.add(Ti.UI.createLabel({ text: title + " View" }));
	var tab = Ti.UI.createTab({
		title: title,
		icon: "/assets/images/tab1.png",
		window: window,
	});
	return tab;
}

var tabGroupSettings = {
	tabs: [createTab("Tab 1"), createTab("Tab 2"), createTab("Tab 3")],
};
if (OS_ANDROID) {
	tabGroupSettings.shiftMode = false;
//	tabGroupSettings.style = Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION;
}
var tabGroup = Ti.UI.createTabGroup(tabGroupSettings);
tabGroup.addEventListener("open", function() {
//	tabGroup.tabs[1].badge = "5";
});
tabGroup.open();
```
